src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -406,17 +409,21 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
         ssMS << RangeToString(", ", "gridtools::define_caches(", "),")(
             multiStage.getCaches(),
             [&](const std::pair<int, Cache>& AccessIDCachePair) -> std::string {
+              auto const& cache = AccessIDCachePair.second;
+              DAWN_ASSERT(cache.getInterval().is_initialized() ||
+                          cache.getCacheIOPolicy() == Cache::local);
+
+              if(cache.getInterval().is_initialized())
```

Comment:
if we already assert on top, should we not just merge these asserts?

Reply:
the problem is  *(cache.getInterval) will seg fault if not initialized

Reply:
but we already asserted if `cache.getInterval().is_initialized()`, so we can do:
```
DAWN_ASSERT(cache.getInterval().is_initialized() || cache.getCacheIOPolicy() == Cache::local);
DAWN_ASSERT(intervalDefinitions.IntervalToNameMap.count(*(cache.getInterval())));
```

Reply:
so say you have is_initialized() == False and cache.getCacheIOPolicy() == Cache::local

The first assert will pass, and second assert will seg fault. While you would like to pass